### PR TITLE
Add Document.addExtensionObject and Document.getExtensionObject

### LINKED ISSUE #252: AttributeError: 'Document' object has no attribute 'getExtensionObject'
etl-to-synbiohub-pipeline uses `Document.getExtensionObject`

Update: also implement `Document.addExtensionObject`

### PR DESCRIPTION
**Pull request by tcmitchell**
Fixes #252

### CODE REVIEW

File: test/test_document.py [hunk 1]
```diff
@@ -396,5 +397,78 @@ def test_add_attachment(self):
...
+        # We need to construct and add an extension object
+        # It should then be accessible via getExtensionObject
+        # We can also peel back the curtain and make sure the object
+        #  is in doc.SBOLObjects and the URI is not in doc.OwnedObjects
+        # Three kinds of things to add:
+        #  1. An existing type, like ComponentDefinition
+        cd = sbol2.ComponentDefinition('cd')
+        doc = sbol2.Document()
+        doc.addExtensionObject(cd)
+        obj = doc.getExtensionObject(cd.identity)
```
Comment (bbartley) [inline test/test_document.py:448]: At this line, what would be the result of...?
```
self.assertEqual(len(doc.componentDefinitions), 1)
```
There might be some janky behavior here, as a result of adding a Core object as an Extension object.

Comment (tcmitchell) [reply test/test_document.py:448]: The assertEqual passes. `len(doc.componentDefinitions) == 1`

Comment (bbartley) [reply test/test_document.py:448]: OK, this is the ideal behavior.  I suggest including that assertion in the test, if there's not already a similar test for the underlying `Document.add` method.  I'm not at all sure pySBOL would actually pass this test.

Comment (tcmitchell) [reply test/test_document.py:449]: I added that assert to the unit test.

**Pull request merged**